src/autodiff.jl

```diff
@@ -319,68 +382,135 @@ function uncompress_kernel!(J::CUSPARSE.CuSparseMatrixCSR, compressedJ, coloring
-        arrays.x[nvbus+1:nvbus+ninj] .= pinj
-        arrays.t1sx .= arrays.x
-        arrays.t1sF .= 0.0
+        J.x[1:nvbus] .= v_m
+        J.x[nvbus+1:nvbus+ninj] .= pinj
+        J.t1sx .= J.x
+        J.t1sF .= 0.0
     else
         error("Unsupported Jacobian structure")
     end
 
-    seed_kernel!(arrays.t1sseeds, arrays.varx, arrays.t1svarx, nbus)
+    seed_kernel!(J.t1sseeds, J.varx, J.t1svarx, nbus)
 
     if isa(type, StateJacobian)
         residual_polar!(
-            arrays.t1sF,
-            view(arrays.t1sx, 1:nvbus),
-            view(arrays.t1sx, nvbus+1:2*nvbus),
+            J.t1sF,
+            view(J.t1sx, 1:nvbus),
+            view(J.t1sx, nvbus+1:2*nvbus),
             ybus_re, ybus_im,
             pinj, qinj,
             pv, pq, nbus
         )
     elseif isa(type, ControlJacobian)
         residual_polar!(
-            arrays.t1sF,
-            view(arrays.t1sx, 1:nvbus),
+            J.t1sF,
+            view(J.t1sx, 1:nvbus),
             v_a,
             ybus_re, ybus_im,
-            view(arrays.t1sx, nvbus+1:nvbus+ninj), qinj,
+            view(J.t1sx, nvbus+1:nvbus+ninj), qinj,
             pv, pq, nbus
         )
     else
         error("Unsupported Jacobian structure")
     end
 
-    getpartials_kernel!(arrays.compressedJ, arrays.t1sF, nbus)
-    uncompress_kernel!(arrays.J, arrays.compressedJ, arrays.coloring)
+    getpartials_kernel!(J.compressedJ, J.t1sF, nbus)
+    uncompress_kernel!(J.J, J.compressedJ, J.coloring)
 
     return nothing
 end
 
+"""
+    residual_hessian_adj_tgt!(H::Hessian,
+                        residual_adj_polar!,
+                        lambda, tgt,
+                        vm, va, ybus_re, ybus_im, pinj, qinj, pv, pq, ref,
+                        nbus)
+
+Update the sparse Jacobian entries using AutoDiff. No allocations are taking place in this function.
+
+* `H::Hessian`: Factory created Jacobian object to update
+* `residual_adj_polar`: Adjoint function of residual
+* `lambda`: Input adjoint, usually lambda from a Langrangian
+* `tgt`: A tangent direction or vector that the Hessian should be multiplied with
+* `vm, va, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus`: Inputs both
+  active and passive parameters. Active inputs are mapped to `x` via the preallocated views.
+"""
+function residual_hessian_adj_tgt!(H::Hessian,
+                             residual_adj_polar!,
+                             lambda, tgt,
+                             v_m, v_a, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus)
+    x = H.x
+    ntgt = length(tgt)
+    nvbus = length(v_m)
+    ninj = length(pinj)
+    t1sx = H.t1sx
+    adj_t1sx = similar(t1sx)
+    t1sF = H.t1sF
+    adj_t1sF = similar(t1sF)
+    x[1:nvbus] .= v_m
+    x[nvbus+1:2*nvbus] .= v_a
+    x[2*nvbus+1:2*nvbus+ninj] .= pinj
+    t1sx .= H.x
+    adj_t1sx .= 0.0
+    t1sF .= 0.0
+    adj_t1sF .= lambda
+    # Seeding
+    nmap = length(H.map)
+    t1sseedvec = zeros(Float64, length(x))
+    for i in 1:nmap
```

Comment:
I would maybe add an `@inbounds` macro there

Reply:
Ok. The bounds checking actually creates a real overhead. So I'm not sure whether to use it everywhere or not.

Reply:
In fact, even if bounds checking creates an overhead, it is actually negligible on modern architecture, as far as I observed